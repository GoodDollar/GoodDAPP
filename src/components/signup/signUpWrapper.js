// @flow
import React, { useCallback } from 'react'
import { SafeAreaView } from 'react-native'
import CustomButton from '../common/buttons/CustomButton'
import Section from '../common/layout/Section'
import Wrapper from '../common/layout/Wrapper'
import { withStyles } from '../../lib/styles'

type FooterProps = {
  valid?: boolean,
  submitText?: string,
  handleSubmit: () => any,
  loading?: boolean,
}

const Footer = ({ valid, submitText, handleSubmit, loading }: FooterProps) => {
  const _handleSubmit = useCallback(handleSubmit)
  return (
    <CustomButton disabled={!valid || loading} loading={loading} onPress={_handleSubmit} style={{ flex: 1 }}>
      {submitText}
    </CustomButton>
  )
}

Footer.defaultProps = {
  submitText: 'Next',
}

const CustomWrapper = (props: any) => {
  const { footerComponent: FooterComponent } = props
  return (
    <SafeAreaView style={props.styles.safeArea}>
      <Wrapper backgroundColor="transparent" style={props.styles.wrapper}>
        <Section grow style={[props.styles.section, props.styles.transparentBackground]}>
          <Section.Stack grow justifyContent="space-evenly">
            {props.children}
          </Section.Stack>
          <Section.Row justifyContent="flex-end">
            {FooterComponent ? <FooterComponent {...props} /> : <Footer {...props} />}
          </Section.Row>
        </Section>
      </Wrapper>
    </SafeAreaView>
  )
}

const getStylesFromProps = ({ theme }) => ({
  safeArea: {
    flex: 1,
  },
  wrapper: {
    padding: theme.sizes.defaultDouble,
  },
  section: {
    padding: 0,
  },
  transparentBackground: {
    backgroundColor: 'transparent',
  },
})

export default withStyles(getStylesFromProps)(CustomWrapper)
