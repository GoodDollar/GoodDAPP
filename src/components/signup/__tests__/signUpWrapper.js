import React from 'react'
import renderer from 'react-test-renderer'
import signUpWrapper from '../signUpWrapper'
import { withThemeProvider } from '../../../__tests__/__util__'

const SignUpWrapperTheme = withThemeProvider(signUpWrapper)

describe('SignUp - signUpWrapper', () => {
  it('matches snapshot', async () => {
    let component

    // eslint-disable-next-line require-await
    await renderer.act(async () => (component = renderer.create(<SignUpWrapperTheme />)))
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
