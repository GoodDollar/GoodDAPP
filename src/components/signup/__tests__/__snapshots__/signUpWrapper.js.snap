// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp - signUpWrapper matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-flex-13awgt0 r-paddingBottom-97e31f r-paddingLeft-bv2aro r-paddingRight-hxflta r-paddingTop-10xqauy"
  >
    <div
      className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-display-6koalj r-flexDirection-eqz5dr r-flexGrow-16y2uox r-maxHeight-87ditf r-overflowX-lltvgl r-overflowY-1rnoaur r-padding-nsbfu8 r-position-bnwqim r-width-13qz1uu"
      data-name="viewWrapper"
    >
      <div
        className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-borderRadius-1f0042m r-flexGrow-16y2uox r-padding-t60dpp r-paddingHorizontal-1e081e0 r-paddingVertical-1yzf0co"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1oszu61 r-flexDirection-eqz5dr r-flexGrow-16y2uox"
          style={
            Object {
              "WebkitBoxPack": "space-evenly",
              "WebkitJustifyContent": "space-evenly",
              "justifyContent": "space-evenly",
              "msFlexPack": "space-evenly",
            }
          }
        />
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
          style={
            Object {
              "WebkitBoxPack": "end",
              "WebkitJustifyContent": "flex-end",
              "justifyContent": "flex-end",
              "msFlexPack": "end",
            }
          }
        >
          <div
            className="css-view-1dbjc4n r-alignItems-1oszu61 r-display-6koalj r-flex-13awgt0 r-justifyContent-1777fci r-padding-t60dpp"
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.12)",
                "borderBottomColor": "rgba(0,175,255,1.00)",
                "borderBottomLeftRadius": "50px",
                "borderBottomRightRadius": "50px",
                "borderBottomStyle": "solid",
                "borderBottomWidth": "0px",
                "borderLeftColor": "rgba(0,175,255,1.00)",
                "borderLeftStyle": "solid",
                "borderLeftWidth": "0px",
                "borderRightColor": "rgba(0,175,255,1.00)",
                "borderRightStyle": "solid",
                "borderRightWidth": "0px",
                "borderTopColor": "rgba(0,175,255,1.00)",
                "borderTopLeftRadius": "50px",
                "borderTopRightRadius": "50px",
                "borderTopStyle": "solid",
                "borderTopWidth": "0px",
                "marginBottom": "0px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
                "minHeight": "44px",
                "minWidth": "64px",
                "paddingLeft": "0px",
                "paddingRight": "0px",
              }
            }
          >
            <div
              aria-disabled={true}
              className="css-view-1dbjc4n r-flex-13awgt0 r-overflow-1udh08x r-position-bnwqim"
              disabled={true}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              style={
                Object {
                  "WebkitAlignSelf": "stretch",
                  "alignSelf": "stretch",
                  "borderBottomLeftRadius": "50px",
                  "borderBottomRightRadius": "50px",
                  "borderTopLeftRadius": "50px",
                  "borderTopRightRadius": "50px",
                  "msFlexItemAlign": "stretch",
                }
              }
            >
              <div
                className="css-view-1dbjc4n r-alignItems-1awozwy r-flex-13awgt0 r-flexDirection-18u37iz r-justifyContent-1777fci"
                style={
                  Object {
                    "WebkitAlignSelf": "stretch",
                    "alignSelf": "stretch",
                    "msFlexItemAlign": "stretch",
                  }
                }
              >
                <div
                  className="css-view-1dbjc4n r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr"
                >
                  <div
                    className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flex-13awgt0 r-flexDirection-18u37iz"
                  >
                    <div
                      className="css-text-901oao r-alignItems-1awozwy r-color-jwli3a r-fontFamily-vnw8o6 r-fontSize-ubezar r-fontWeight-majxgm r-justifyContent-1777fci r-letterSpacing-oxtfae r-lineHeight-hbpseb r-paddingTop-njp1lv r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-tsynxw"
                      dir="auto"
                    >
                      Next
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
